javatests/arcs/core/storage/CapabilitiesResolverTest.kt

```diff
@@ -38,71 +45,71 @@ class CapabilitiesResolverTest {
         CapabilitiesResolver.defaultCreators.clear()
     }
 
+    fun verifyStorageKey(key: StorageKey?, expectedClass: Class<out StorageKey>) {
+        assertThat(key).isInstanceOf(ReferenceModeStorageKey::class.java)
+        assertThat((key as ReferenceModeStorageKey).backingKey).isInstanceOf(expectedClass)
+        assertThat(key.storageKey).isInstanceOf(expectedClass)
+    }
+
     @Test
     fun capabilitiesResolver_createsStorageKeysDefault() {
-        val options = CapabilitiesResolver.StorageKeyOptions(ArcId.newForTest("test"))
+        val options = CapabilitiesResolver.CapabilitiesResolverOptions(ArcId.newForTest("test"))
         val resolver = CapabilitiesResolver(options)
         assertThat(resolver.findStorageKeyProtocols(Capabilities.TiedToArc))
             .containsExactly(VOLATILE_DRIVER_PROTOCOL)
         assertThat(resolver.findStorageKeyProtocols(Capabilities.TiedToRuntime)).isEmpty()
         assertThat(resolver.findStorageKeyProtocols(Capabilities.Persistent)).isEmpty()
-        assertThat(resolver.createStorageKey(Capabilities.TiedToArc))
-            .isInstanceOf(VolatileStorageKey::class.java)
+        verifyStorageKey(resolver.createStorageKey(Capabilities.TiedToArc, thingSchema, handleId), VolatileStorageKey::class.java)
         assertThrows(IllegalArgumentException::class) {
-            resolver.createStorageKey(Capabilities.TiedToRuntime)
+            resolver.createStorageKey(Capabilities.TiedToRuntime, thingSchema, handleId)
         }
         assertThrows(IllegalArgumentException::class) {
-            resolver.createStorageKey(Capabilities.Persistent)
+            resolver.createStorageKey(Capabilities.Persistent, thingSchema, handleId)
         }
         assertThrows(IllegalArgumentException::class) {
             resolver.createStorageKey(Capabilities(setOf(
                 Capabilities.Capability.TiedToArc,
                 Capabilities.Capability.Persistent
-            )))
+            )), thingSchema, handleId)
         }
     }
 
     @Test
     fun capabilitiesResolver_createsStorageKeysCtor() {
-        val options = CapabilitiesResolver.StorageKeyOptions(ArcId.newForTest("test"))
+        val options = CapabilitiesResolver.CapabilitiesResolverOptions(ArcId.newForTest("test"))
         val ramDiskCreator: Pair<Capabilities, StorageKeyCreator> = Capabilities.TiedToRuntime to
-            { storageKeyOptions, _ -> RamDiskStorageKey(storageKeyOptions.arcId.toString()) }
+            { storageKeyOptions -> RamDiskStorageKey(storageKeyOptions.arcId.toString()) }
         val resolver = CapabilitiesResolver(
             options,
             mutableMapOf(RAMDISK_DRIVER_PROTOCOL to ramDiskCreator)
```

Comment:
Why do it this way instead of referencing RamDisk which will trigger it to automatically register the provider?

Reply:
for tests, i want to be able to turn it on and off